src/main/java/com/parasoft/findings/jenkins/coverage/ParasoftCoverageStep.java

```diff
@@ -0,0 +1,108 @@
+package com.parasoft.findings.jenkins.coverage;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.*;
+import io.jenkins.plugins.coverage.metrics.steps.CoverageStep;
+import io.jenkins.plugins.coverage.metrics.steps.CoverageTool;
+import io.jenkins.plugins.util.AbstractExecution;
+import io.jenkins.plugins.util.LogHandler;
+import io.jenkins.plugins.util.RunResultHandler;
+import org.apache.commons.lang3.StringUtils;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.steps.Step;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor;
+import org.jenkinsci.plugins.workflow.steps.StepExecution;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Set;
+
+public class ParasoftCoverageStep extends Step implements Serializable {
+    private static final long serialVersionUID = -2235239576082380147L;
+    private String pattern = StringUtils.EMPTY;
+    private static final String DEFAULT_PATTERN = "**/coverage.xml";
```

Comment:
Maybe we can change the access modifier of these constants in coverage recorder to package-private(no modifier) and use them in this class, then we don't need to define them here.